From 62d6feb52563909853e62e4e6cbf8b8c1d6fc114 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 15:53:25 +0800
Subject: [PORT FROM MAIN] libstagefright: Do not render buffer whose length equals 0 even when EOS

BZ: 52860

when Acoder encourters EOS and the returned buffer from HAL length equals 0,
it will still render the buffer.the empty buffer may not filled with any data
and so that we see the display is green.Don't queue the buffer to nativewindow if
the buffer length is 0 even it 's EOS

Change-Id: I50adc6a27f9cb51cceec0cce715f9a2b254de5b7
Orig-Change-Id: I8063e10dc176af99a6f1d9e382ceddc935e0881c
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/ACodec.cpp |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index a2c8d4c..7faf052 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -3112,6 +3112,10 @@ bool ACodec::BaseState::onOMXFillBufferDone(
                     mCodec->mSentFormat = true;
                 }
             }
+
+            if (mCodec->mNativeWindow != NULL && rangeLength == 0) {
+                reply->setInt32("hide",true);
+            }
             mCodec->mFirstFrame = false;
 
             notify->setMessage("reply", reply);
@@ -3167,6 +3171,7 @@ void ACodec::BaseState::onOutputBufferDrained(const sp<AMessage> &msg) {
     // need to apply the format change regardless as we would get
     // only one format change call, associated with a buffer.
     int32_t sentformat;
+    int32_t hide;
     sp<AMessage> reply =
         new AMessage(kWhatOutputBufferDrained, mCodec->id());
 
@@ -3178,7 +3183,8 @@ void ACodec::BaseState::onOutputBufferDrained(const sp<AMessage> &msg) {
 
     if (mCodec->mNativeWindow != NULL
             && msg->findInt32("render", &render) && render != 0
-            && (info->mData == NULL || info->mData->size() != 0)) {
+            && (info->mData == NULL || info->mData->size() != 0)
+            && !msg->findInt32("hide", &hide)) {
         // The client wants this buffer to be rendered.
 
         status_t err;
-- 
1.7.4.1

